common/tier4_autoware_utils/include/tier4_autoware_utils/ros/polling_subscriber.hpp

```diff
@@ -30,24 +30,36 @@ class InterProcessPollingSubscriber
   std::optional<T> data_;
 
 public:
-  explicit InterProcessPollingSubscriber(rclcpp::Node * node, const std::string & topic_name)
+  explicit InterProcessPollingSubscriber(
+    rclcpp::Node * node, const std::string & topic_name, const rclcpp::QoS & qos = rclcpp::QoS{1})
   {
     auto noexec_callback_group =
       node->create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive, false);
     auto noexec_subscription_options = rclcpp::SubscriptionOptions();
     noexec_subscription_options.callback_group = noexec_callback_group;
 
     subscriber_ = node->create_subscription<T>(
-      topic_name, rclcpp::QoS{1},
+      topic_name, qos,
       [node]([[maybe_unused]] const typename T::ConstSharedPtr msg) { assert(false); },
       noexec_subscription_options);
+    if (qos.get_rmw_qos_profile().depth > 1) {
+      RCLCPP_WARN(
+        node->get_logger(),
+        "InterProcessPollingSubscriber will be used with depth > 1, which may cause inefficient "
+        "serialization while updateLatestData()");
+    }
```

Comment:
I think this warning is not necessary since now the InterProcessPollingSubscriber can handle the topic whose depth is not 1.

Reply:
The user have the freedom to use this subscriber with depth > 1, but eventually he/she will not use the stacked obsolete data because `getData` returns only the latest data.

The purpose of this subscriber is to minimize the repeated serialization overheads in `subscriber_->take(tmp, message_info)`, so if this warning appears then he's not using this as expected. This warning is printed only once in the constructor.